pysmt/test/test_solving.py

```diff
@@ -122,6 +122,38 @@ def test_get_model_sat(self):
             self.assertTrue(res.get_value(varX) == Real(8))
 
     @skipIfNoSolverForLogic(QF_BOOL)
+    def test_get_implicant_unsat(self):
+        varA = Symbol("A", BOOL)
+        varB = Symbol("B", BOOL)
+
+        f = And(varA, Not(varB))
+        g = f.substitute({varB:varA})
+
+        res = get_implicant(g, logic=QF_BOOL)
+        self.assertIsNone(res, "Formula was expected to be UNSAT")
+
+        for solver in get_env().factory.all_solvers(logic=QF_BOOL):
+            res = get_implicant(g, solver_name=solver)
+            self.assertIsNone(res, "Formula was expected to be UNSAT")
+
+    @skipIfNoSolverForLogic(QF_LRA)
+    def test_get_implicant_sat(self):
+        varA = Symbol("A", BOOL)
+        varX = Symbol("X", REAL)
+
+        f = And(varA, Equals(varX, Real(8)))
+
+        res = get_implicant(f, logic=QF_LRA)
```

Comment:
Same here. This is repeated below. Since the test is decorated with skipIfNoSolverForLogic, we are guaranteed that the for below will be entered.